### ISSUE TITLE
Waveguide input / output device benchmark problems

### ISSUE THREAD

**Issue by ianwilliamson** [NanoComp/photonics-opt-testbed#6]
We can contribute the design patterns and code for the waveguide input-output devices shown in our [Inverse design of photonic devices with strict foundry fabrication constraints](https://arxiv.org/abs/2201.12965) paper. These results use the [Ceviche](https://github.com/fancompute/ceviche/) simulator, but rely on additional modules that we developed to set up the device geometries and calculate scattering parameters. We can open source these modules separately from this repository, but introduce a script here for users to run them with selected designs.

## TODO

- [ ] Open source the code that sets up the device geometries, runs the simulations, and calculates the scattering parameters
- [ ] Contribute design patterns from the paper to this testbed repository

## Broadband waveguide bend

<img width="218" alt="image" src="https://user-images.githubusercontent.com/8869582/160657534-594a33a4-70a9-408f-ac4d-3d8870c472bb.png">

<img width="356" alt="image" src="https://user-images.githubusercontent.com/8869582/160659108-7b96bc38-15af-46f7-a360-5fe829ee9773.png">

## Mode converter

<img width="271" alt="image" src="https://user-images.githubusercontent.com/8869582/160657581-28803e46-8119-4b83-ae34-ba1fc3e294c8.png">

<img width="413" alt="image" src="https://user-images.githubusercontent.com/8869582/160659253-55f54d83-33fa-41dc-a953-c67659095094.png">

## Beam splitter

<img width="338" alt="image" src="https://user-images.githubusercontent.com/8869582/160657612-9d459761-3678-423f-91e9-6280015454cb.png">

## Wavelength division multiplexer

<img width="655" alt="image" src="https://user-images.githubusercontent.com/8869582/160657657-24a5e252-293e-44e4-a38a-b34e3d1e59fa.png">

<img width="812" alt="image" src="https://user-images.githubusercontent.com/8869582/160659307-8c3a6c86-047d-497f-8e89-c88922d6fe08.png">

**Comment by ianwilliamson** [comment]
> Let's start with just the mode converter?

Sure. In this repo we can add an "end point" for whichever component we want.

**Comment by Dj1312** [comment]
> We can contribute the design patterns and code for the waveguide input-output devices shown in our [Inverse design of photonic devices with strict foundry fabrication constraints](https://arxiv.org/abs/2201.12965) paper. These results use the [Ceviche](https://github.com/fancompute/ceviche/) simulator, but rely on additional modules that we developed to set up the device geometries and calculate scattering parameters. We can open source these modules separately from this repository, but introduce a script here for users to run them with selected designs.
> ## TODO
> 
>     * [ ]  Open source the code that sets up the device geometries, runs the simulations, and calculates the scattering parameters
> 
>     * [ ]  Contribute design patterns from the paper to this testbed repository

@ianwilliamson if you can share the code of your paper or at least some part to implement the graph of your paper / geometries, that would be awesome !! 🥳

**Comment by Dj1312** [comment]
Thanks for the sharing @ianwilliamson !

**Comment by Dj1312** [comment]
Yes ! I just found out, that is why my last message was a thanks.
